static/js/message_list.js

```diff
@@ -121,7 +135,7 @@ export class MessageList {
     }
 
     can_mark_messages_read() {
-        return this.data.can_mark_messages_read();
+        return !this.reading_prevented && this.data.can_mark_messages_read();
```

Comment:
Reordered this with a block comment to make it easier to trace what we're doing.